src/file.cpp

```diff
@@ -375,7 +375,12 @@ namespace libtorrent
 		{
 			std::string sep_path { path };
 			std::replace(sep_path.begin(), sep_path.end(), '/', '\\');
-			prepared_path = "\\\\?\\" + (is_complete(sep_path) ? sep_path : combine_path(current_working_directory(), sep_path));
+			if (!is_complete(sep_path))
+			{
+				std::string cwd_path { current_working_directory() };
+				sep_path = combine_path(cwd_path, sep_path);
+			}
+			prepared_path = "\\\\?\\" + sep_path;
```

Comment:
I think the new version here is objectively worse than the previous. Primarily because it's longer, but it also introduces a mutable variable. I take it the problem with the current version is that ``combine_path()`` is being called with one argument that's an rvalue string, and string_view won't construct from that anymore.

what I tried to ask in the comments earlier was, is that constructor deleted or just explicit? would this work?

``combine_path(string_view(current_working_directory()), sep_path)``

If so, I think that would be preferable, as it's optimal.